dts/bindings/usb-c/usb-c-connector.yaml

```diff
@@ -137,31 +168,107 @@ properties:
+            vSafe5V (e.g., 12V) to provide full functionality.
+
+    fr-swap-reqd-type-c-current-as-initial-source:
+        type: int
+        enum: [0, 1, 2, 3]
+        description: |
+            This field is used to determine if the Qualifying Product supports
+            Fast Role Swap in initial Source role.
+
+            In a Sink_Capabilities message there is a field called Fast Role
+            Swap USB Type-C Current, in which the initial Source describes
+            whether it supports Fast Role Swap to become a new Sink, and if so,
+            what current it requires as a new Sink.
+            Valid options are:
+            0 - FR_Swap not supported
+            1 - Default USB Power
+            2 - 1.5A @ 5V
+            3 - 3A @ 5V
+
+    num-snk-pdos:
+        type: int
+        description: |
+            The number of Sink Power Data Objects (PDOs) returned by the Port
+            in a Sink_Capabilities message.
+            Valid values are in the range 1 - 7.
+
+            If epr-supported-as-snk is set to NO, then this field shall be
+            equal to the number of Sink PDOs defined for this Port.
+
+            If epr-supported-as-snk is set to YES, then this field shall be
+            strictly less than the number of Sink PDOs defined for this Port.
+
+    dr-swap-to-dfp-supported:
+        type: boolean
+        description: |
+            Set to YES if Qualifying Product can respond with an Accept to a
+            DR_Swap request to switch from a UFP to a DFP.
+
+    dr-swap-to-ufp-supported:
+        type: boolean
+        description: |
+            Set to YES if Qualifying Product can respond with an Accept to a
+            DR_Swap request to switch from a DFP to a UFP.
+
+    vconn-swap-to-on-supported:
+        type: boolean
+        description: |
+            Set to YES if Qualifying Product can respond with an Accept to a
+            VCONN_Swap message requesting it to start sourcing VCONN.
+
+    vconn-swap-to-off-supported:
+        type: boolean
+        description: |
+            Set to YES if Qualifying Product can respond with an Accept to a
+            VCONN_Swap message requesting it to stop sourcing VCONN.
+
+    accepts-pr-swap-as-src:
+        type: boolean
+        description: |
+            Set to YES if Port ever supports a PR_Swap request while operating
+            as a Source.
+
+    accepts-pr-swap-as-snk:
+        type: boolean
+        description: |
+            Set to YES if Port ever supports a PR_Swap request while operating
+            as a Sink.
+
+    requests-pr-swap-as-src:
+        type: boolean
+        description: |
+            Set to YES if the Qualifying Product may send a PR_Swap request
+            while operating as a Source.
+
+    requests-pr-swap-as-snk:
+        type: boolean
+        description: |
+            Set to YES if the Qualifying Product may send a PR_Swap request
+            while operating as a Sink.
```

Comment:
Some of these properties describe behaviors that will sometimes vary according to runtime circumstances or policy (and thus should probably not be constant port properties).

For instance, whether a port accepts a PR Swap may depend on the battery level or what is plugged into the other ports. FR Swap required current may depend on the power state of the device.

Reply:
I beleive this property is capturing  whether it may send PR_Swap request. I mean it need not to send request consistently, but if it could anytime it's true, kind of.